app/services/page_updater.rb

```diff
@@ -0,0 +1,92 @@
+class PageUpdater
+
+  def initialize(page, page_url=nil)
+    @page = page
+    @page_url = page_url
+  end
+
+  def update(params)
+    @params, @errors, @refresh = params, {}, false
+    update_plugins()
+    update_shares()
+    update_page()
+    @errors.empty?
+  end
+
+  def errors
+    @errors
+  end
+
+  def refresh?
+    @refresh || false
```

Comment:
but I want the method to return a boolean value cause it ends in a question mark